### DISCUSSION TITLE
Trying to send HTTP from pb_hooks whenever new user record is created

### DISCUSSION THREAD

**Discussion by natecasp** [pocketbase/pocketbase#5327] [Q&A] [answered]
Hi all, 

I want to send an HTTP request to a webhook whenever a new user record is created.
The data I want to send is an array of all of the user records, with only the "username" column/field.

Should be pretty straight forward, but when I try to implement it, the HTTP Post is firing for every single record. I want it to fire just once with an array containing the "username" from every record in the collection.

Currently happening:
HTTP Post #1:
[{"username":"user_1"}]

HTTP Post #2:
[{"username":"user_2"}]
etc.

What I want to output:
HTTP Post:
[{"user_1","user_2",...}]

Here is the hook as it is currently set up. Can anyone help me with where I am going wrong here?
Thanks!
```
onRecordAfterCreateRequest((e) => {
    
	const result = arrayOf(new DynamicModel({
		// describe the shape of the data (used also as initial values)
		"username": ""
	}))
		
	$app.dao().db().select("username").from("users").all(result) 
	
	const res = $http.send({
			url:     "https://hook.us1.make.com/ur4t7wq8t6o9klkxvq3xcgafv2jj96g8",
			method:  "POST",
			body:    JSON.stringify(result), 
			headers: {"content-type": "application/json"},
			timeout: 120, // in seconds
		})
}, "users")
```

**Comment by ganigeorgiev** [comment]
The wanted output seems to be invalid json but I assume that you want to send an array of usernames, aka. `["user_1", "user_2"]`.

In that case you can call `map` on the result and extract the username:

```js
body: JSON.stringify(result.map((item) => item.username)
```

Another alternative could be instead of fetching array of `DynamicModel`s to retrieve only the column as slice of strings, aka:

```js
const result = arrayOf("")
$app.dao().db().select("username").from("users").column(result) 
```

**Reply by natecasp** [reply]
Yes you are correct that I am looking for an array of usernames. I tried both of the proposed results and in both instances the output is creating an array that splits each character of the username like so:
[
    {
        "0": "T",
        "1": "h",
        "2": "i",
        "3": "s",
        "4": "_",
        "5": "I",
        "6": "s",
        "7": "_",
        "8": "A",
        "9": "_",
        "10": "T",
        "11": "e",
        "12": "s",
        "13": "t"
    }
]

Additionally, the HTTP request is still firing for every single record, whereas I only want to to fire once with an array containing the username of every record.

**Reply by ganigeorgiev** [reply, marked as answer]
Both options send the correct payload for me. You are probably executing something else that you didn't share but I can't help you without seeing the code.

Here is an example hook that will send request after user create to verify the correct payload:

```js
onRecordAfterCreateRequest((e) => {
    const result = arrayOf("");

    $app.dao().db().select("username").from("users").column(result)

    const res = $http.send({
        url:     "https://enyvm1md6tt2s.x.pipedream.net",
        method:  "POST",
        body:    JSON.stringify(result),
        headers: {"content-type": "application/json"},
        timeout: 120, // in seconds
    })
}, "users")
```

The sent body is `["users57884","users85572"]`. You can inspect the received body in the history of https://public.requestbin.com/r/enyvm1md6tt2s/2kLGn0dfvAz6qw1C6AA8qygWhmV.


> Additionally, the HTTP request is still firing for every single record, whereas I only want to to 
fire once with an array containing the username of every record.

I don't understand what do you mean by this, neither it is clear when is expected the above request to be triggered (_it also seems to contradict with "I want to send an HTTP request to a webhook whenever a new user record is created."_).  `onRecordAfterCreateRequest` works correctly and fires for every "users" create API request.

I've written this before but please don't treat the the discussions as chatgpt and try to clearly elaborate on your use case to understand the bigger picture and what you are trying to do (see also [XY problem](https://xyproblem.info/)).

_edit: updated to use username instead of id in the above example._

**Reply by natecasp** [reply]
Thanks @ganigeorgiev, after digging into the problem my webhook was the issue and not your proposed solution. What I mean by firing for every record is that my webhook was receiving every record from the query as an individual http post request for some reason and I believed it was due to an error in my code.
